### PR TITLE
APPSEC-1124: S6689+S6690 Align GitHub and GitLab patterns and add rspec examples

### PR DESCRIPTION
**Pull request by daniel-teuchert-sonarsource**

### CODE REVIEW

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gitlab.yaml [hunk 1]
```diff
@@ -22,13 +22,26 @@ provider:
               - glpat-
 
         matching:
-          pattern: "\\b(glpat-[a-zA-Z0-9_-]{20})\\b"
+          pattern: "\\b(glpat-(?<rand>[a-zA-Z0-9_-]{20}))\\b"
         post:
           statisticalFilter:
-            threshold: 3.9 # Based on significant sampling, 4.0 < entropy < 4.8
+            threshold: 3.2 # Based on significant sampling, 3.3 < entropy < 4.3
```
[the remark this answers is not in the document]
Comment (daniel-teuchert-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gitlab.yaml:28]: Yes, it does. The threshold is always smaller than the minimum entropy. Should I add this explanation to the comment?

[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gitlab.yaml:28]: This is what I mean:

```python
>>> 3.3 < 3.2
False
>>> 3.2 < 4.3
True
```

**Review by loris-s-sonarsource — CHANGES_REQUESTED** [review]
3.3 < 3.2 < 4.3 is incorrect

File: sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gitlab.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (daniel-teuchert-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gitlab.yaml:28]: but `threshold` != `entropy`
What the comment means is:
the entropy of this "pattern" is between 3.3 and 4.3.
And just to make sure we will set the entropy threshold a little bit lower than the minimum entropy (so to 3.2).

[the remark this answers is not in the document]
Comment (daniel-teuchert-sonarsource) [reply sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/gitlab.yaml:28]:
```
>>> 'threshold' == 'entropy'
False
```

**Review by loris-s-sonarsource — APPROVED** [review]
```python
>>> 'review' == 'LGTM'
True
```

**Pull request merged**